### PR TITLE
flightcheck: port BL-014 network validator + Workday REST diagnostic + contributor guardrails

### PR DESCRIPTION
**Pull request by johnguy0**
## Description

Port of 5 commits from [`JohnQuinn-Dev/ess-preflight-validator`](https://github.com/JohnQuinn-Dev/ess-preflight-validator/commits/master/) (since 2026-05-01) into this repo's Python FlightCheck framework.

Source commits → ported as:

| Source SHA | Source title | Lands here as |
|---|---|---|
| [`9ed2055`](https://github.com/JohnQuinn-Dev/ess-preflight-validator/commit/9ed2055) | feat: BL-014 Network Connectivity validator + integration selection | **Commit 1** — `checks/network.py`, `checks/firewall_export.py`, `config/required-endpoints.json`, lazy-auth gate in `cli.py`, new "Vendor TCP/HTTPS reachability" tier-registry row |
| [`5eb19bc`](https://github.com/JohnQuinn-Dev/ess-preflight-validator/commit/5eb19bc) + [`44ac58d`](https://github.com/JohnQuinn-Dev/ess-preflight-validator/commit/44ac58d) | Add REST Endpoints Pre-flight Validation Script (PR #3 by @is-goutham) | **Commit 2** — `scripts/diagnostics/test_workday_rest_endpoints.py` (standalone Python OAuth diagnostic) + `WD-REST-MANUAL` checklist entry in `checks/workday.py` + tier-registry note |
| [`70065b9`](https://github.com/JohnQuinn-Dev/ess-preflight-validator/commit/70065b9) | Add SSO test flow template (WIP) | **Commit 2** — vendored to `src/reference/workday-sso-test-flow/` with adapted README |
| [`2ac16e9`](https://github.com/JohnQuinn-Dev/ess-preflight-validator/commit/2ac16e9) | Add contributor guardrails: PR template + pre-PR checklist | **Commit 3** — merged into existing `.github/PULL_REQUEST_TEMPLATE.md` + `CONTRIBUTING.md`, translated to this repo's conventions |

## Related issue

Refs the BL-014 backlog item from the source repo (no equivalent issue tracked here yet — happy to file one if maintainers want it for changelog purposes).

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update

## Testing

```text
pytest tests/ -q     → 133 passed, 8 skipped (baseline was 103; +30 new tests, 0 regressions)
```

New test modules:
- `tests/flightcheck/checks/test_network.py` (15 tests covering the 6 failure modes + selection/config edge cases)
- `tests/flightcheck/checks/test_firewall_export.py` (7 tests, golden-file markdown render)
- `tests/flightcheck/test_cli_lazy_auth.py` (10 tests pinning per-scope auth requirement)

Smoke-tested in a temp working dir with a minimal `.local/config.json`:
- `cli.py --export-firewall-requirements` → writes `workspace/flightcheck/firewall-requirements.md`, **no Microsoft auth attempted**
- `cli.py --scope network` → runs with `Skipping Microsoft auth (not required for --scope network)` banner, executes probes
- `diagnostics/test_workday_rest_endpoints.py --help` → parses cleanly

Full E2E run of the Workday REST diagnostic requires a customer Workday tenant + an OAuth API Client, which is the documented design — see `scripts/diagnostics/README.md`.

## Key design decisions

After the [rubber-duck pass](https://github.com/github/copilot-cli) caught 10 issues in the initial plan, the final design:

1. **Workday REST went to a standalone diagnostic, not an in-runner FlightCheck check.** The source uses OAuth 2.0 Authorization Code with a browser-opened login — that's Workaround #1 explicitly rejected for runtime FlightCheck checks in `tests/fixtures/cassettes/INDEX.md`'s "Workday WQL config-validation pattern" section ("doesn't escape the chicken-and-egg"). The source repo didn't *solve* this either — it accepts the customer doing the OAuth API Client registration as a prerequisite. Faithful port = same interactive UX, just in Python.
2. **Network reachability is the exception to the cardinal rule.** TCP/HTTPS reachability probes don't consume vendor API response contracts, so the validated/validatable/documented tier system doesn't bind. Documented this explicitly with a new "Vendor TCP/HTTPS reachability" row in the API tier registry so future readers don't have to re-derive the exemption.
3. **Lazy-auth gate via `_requires_microsoft_auth(scope)`.** Without it, `--scope network` would trigger an interactive MSAL prompt before any probe ran — defeating the shift-left value. Mirrors the existing PVA scope gate at `cli.py:162` (which is the precedent).
4. **Injectable `TcpProber` / `HttpsProber` protocol, not `responses`/`respx`.** Those libraries only mock HTTP-level traffic and can't fake `socket.create_connection` refusal. A protocol abstraction is the cleanest testable seam.
5. **Uses `requests` (not `httpx`).** `httpx` is NOT in `solutions/ess-maker-skills/scripts/requirements.txt` — verified before coding. `requests` is, so we stay on it.
6. **Default OAuth UX is paste-the-redirect-URL.** Spinning up a TLS-terminating localhost server would require shipping a self-signed cert (trust-store friction). The source PowerShell does paste; we match. Optional `--listen` flag activates a stdlib `http.server` loopback only when `--redirect-uri` is `http://localhost...` (advanced customers who registered an HTTP redirect URI).
7. **`WD-REST-MANUAL` is conditional.** Only fires when `_workday_flows` was already detected — non-Workday customers see no noise.
8. **Secrets and PII hygiene.** OAuth client secret / authorization code / access token / refresh token are never logged. Token-endpoint errors reduced to status + error class. Employee PII redacted in JSON output by default; `--include-pii` opt-in.

## Pre-PR Checklist (REQUIRED)

- [x] **Rebased on latest `main`** — branched from `887dd41`, no rebase needed
- [x] **Files changed tab reviewed** — 13 files (8 new, 5 modified); no accidental deletions
- [x] **Local lint / tests pass** — `pytest tests/ -q` clean; `cli.py --help` parses
- [x] **No references to files outside the repo** — N/A
- [x] **Defaults match repo conventions** — output paths default to `workspace/flightcheck/`; write tests in the Workday REST diagnostic are opt-in via `--include-write-tests`
- [x] **FlightCheck integration** — new `network` scope wired into `SCOPE_MAP` and `FULL_SCOPE`; tests under `tests/flightcheck/checks/`; new `test_cli_lazy_auth.py` pins the scope-to-auth mapping
- [x] **API tier registry honored** — new "Vendor TCP/HTTPS reachability" row added with rationale; existing Workday WQL/REST row updated with the standalone-diagnostic note
- [x] **Docs updated** — `tests/fixtures/cassettes/INDEX.md`, new `scripts/diagnostics/README.md`, new `src/reference/workday-sso-test-flow/README.md`, `cli.py` module docstring scopes list

## Checklist

- [x] My code follows the existing style
- [x] I have added/updated tests where applicable
- [x] I have updated documentation as needed

## Notes for reviewer

- The 3 commits are independently reviewable in order. Commit 1 (network + lazy-auth) is the only one that touches `cli.py`; commit 2 is the largest by line count (mostly the diagnostic + its README); commit 3 is small docs-only.
- The new `WD-REST-MANUAL` `NotConfigured` checkpoint will show up in any FlightCheck run scope that includes Workday — by design (it's a deployment checklist item). Let me know if you'd rather scope it tighter (e.g. only `--scope publishing`).
- Source's `CHANGELOG.md` / `VERSION.txt` housekeeping from commit `9ed2055` is out of scope (this repo isn't versioned the same way; no top-level CHANGELOG). Happy to add one if maintainers want it.
- Source's `ESS-Validator-Session.ps1` interactive integration-selection menu is replaced by the config-driven `network.integrations` / `network.servicenow_instance` keys in `.local/config.json` — fits this repo's non-interactive CLI model.

### CODE REVIEW

File: solutions/ess-maker-skills/scripts/flightcheck/checks/network.py [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+
+    # Aggregate: one CheckResult per integration; per-host detail in the result field.
+    reachable_count = sum(
+        1 for _, t, h in probe_results
+        if t.status == ProbeStatus.REACHABLE and h.status == ProbeStatus.REACHABLE
+    )
+    warning_count = sum(
+        1 for _, t, h in probe_results
+        if t.status == ProbeStatus.REACHABLE
+        and h.status in (ProbeStatus.HTTP_5XX, ProbeStatus.TLS_ERROR)
```
Comment (nehaoss) [inline solutions/ess-maker-skills/scripts/flightcheck/checks/network.py:310]: **Bug: Incomplete HTTPS failure mode counting causes false 'Passed' status**

`warning_count` only counts `HTTP_5XX` and `TLS_ERROR` HTTPS outcomes, but `_RequestsHttpsProber.probe()` can also return `TIMEOUT` (from `RequestsTimeout`, line ~147) and `REFUSED` (from `RequestsConnectionError`, line ~150).

When TCP succeeds but HTTPS returns either of these, the endpoint isn't counted in any of the three aggregation categories (`reachable_count`, `warning_count`, or `failed_count`), so the status falls through to `Passed`. The detail text *does* print `[WARN]` for these cases (line ~324), creating contradictory output.

**Suggested fix:**
```python
and h.status in (ProbeStatus.HTTP_5XX, ProbeStatus.TLS_ERROR, ProbeStatus.TIMEOUT, ProbeStatus.REFUSED)
```

**Review by nehaoss — COMMENTED** [review]
Nice work on the network validator port. The OAuth flow, PII redaction, auth gating, firewall export, and test coverage are well-structured. Found one counting bug in the HTTPS failure aggregation.

**Review by nehaoss — COMMENTED** [review]
Large PR (2603 additions across 16 files) — porting a network validator + Workday REST diagnostic + contributor guardrails. Comments:

1. **Workday REST diagnostic standalone design**: The rationale for placing this outside FlightCheck (chicken-and-egg OAuth problem) is well documented in the README. The decision to treat HTTP 400/422 as PASS for write endpoints is pragmatic — endpoint reachability is the goal, not business logic.

2. **Network check (\
etwork.py\)**: At 407 lines this is a substantial new check. Please confirm it's covered by the 381-line test file (\	est_network.py\) — ratio looks healthy but verify all branches are exercised.

3. **\
equired-endpoints.json\ config**: Externalizing the endpoint list as JSON is a good separation of concerns. Makes it easy for customers to audit/extend without reading Python.

4. **\	est_cli_lazy_auth.py\** (85 additions): Good to test that the CLI doesn't eagerly authenticate when running checks that don't need it — performance and UX improvement.

5. **Scope concern**: 16 files across diagnostics, flightcheck checks, tests, fixtures, reference docs, and PR template changes. Consider whether this could be split into 2-3 smaller PRs for easier review (e.g., network validator separate from Workday REST diagnostic separate from contributor guardrails). If not, consider a more detailed PR description mapping each file to its purpose.

6. **PR template changes overlap with PR #105**: Both PRs modify \.github/PULL_REQUEST_TEMPLATE.md\. Whoever merges second will have a conflict — coordinate merge order.

**Comment by nehaoss** [comment]
👋 Friendly reminder: This PR has been inactive for a while. Could the author or reviewers take a look and either push it forward, request changes, or close it if it's no longer needed? Let's keep our PR queue healthy! 🚀
